csrc/scheduler/matmul_utils.cpp

```diff
@@ -296,6 +270,44 @@ std::string isMatmulFusionDefinitionSupported(
     }
   }
 
+  // Check that canonical dim order is (B)MNK
+  // TODO: Remove this check once we are confident that non-standard orders are
+  // properly handled
+  {
+    std::vector<ValGroup> dim_ordering = mma_utils::canonicalDimOrdering(
+        tensor_roles, id_roles, permissive_graph);
+    VectorOfUniqueEntries<MatmulDomain> role_order;
+    for (const ValGroup& g : dim_ordering) {
+      const auto it = id_roles.find(g);
+      NVF_ERROR(it != id_roles.end());
+      role_order.pushBack(it->second);
+    }
+    NVF_ERROR(
```

Comment:
So `scheduleMatmul` can now partially support having multiple M/N dims, but we are still rejecting it in the scheduler for now.

Reply:
Wait, should this be the following?
```C++
if (role_order.size() != 3 && role_order.size() != 4) {
  return "Expected either….";
}
```

Reply:
> Wait, should this be the following?

Ah! Good catch!

Reply:
> So `scheduleMatmul` can now partially support having multiple M/N dims, but we are still rejecting it in the scheduler for now?

This check will not complain if there are multiple M or N dims, since we build `role_order` as the ordering of the _roles_ not all the dimensions that constitute all the roles. For example we might have A[M1, K1, M2] and B[K1, N1] and ordering [M, N, K].

Reply:
> > So `scheduleMatmul` can now partially support having multiple M/N dims, but we are still rejecting it in the scheduler for now?
> 
> This check will not complain if there are multiple M or N dims, since we build `role_order` as the ordering of the _roles_ not all the dimensions that constitute all the roles. For example we might have A[M1, K1, M2] and B[K1, N1] and ordering [M, N, K].

But wouldn't the `role_order.size()` check reject the fusion?

Reply:
No, `role_order` would be [M, N, K] in that case. It doesn't hold dimensions, but roles.